src/stages/stage5/timeline.c

```diff
@@ -334,197 +135,713 @@ static int stage5_magnetto(Enemy *e, int t) {
-		}
+TASK(lightburst_fairies_2, {
+	int num;
+	cmplx pos1;
+	cmplx pos2;
+	cmplx exit;
+}) {
+	for (int i = 0; i < ARGS.num; i++) {
+		cmplx pos = ARGS.pos1 + ARGS.pos2 * i;
+		INVOKE_TASK(lightburst_fairy_2,
+			.pos = pos,
+			.move_enter = move_towards(pos + ARGS.exit * 70 , 0.05),
+			.move_exit = move_linear(ARGS.exit)
+		);
+		WAIT(40);
 	}
+}
 
-	AT(2900)
-		global.boss = stage5_spawn_midboss();
+TASK(loop_swirl_move, {
+    MoveParams *move;
+	cmplx turn;
+	cmplx retention;
+}) {
+	int begin = creal(ARGS.turn) + 1;
+	int end = cimag(ARGS.turn) - 1;
+	int time = end - begin;
+
+	ARGS.move->retention = ARGS.retention;
+	WAIT(time / 1.5);
+	ARGS.move->retention = 1;
+}
 
-	AT(2920) {
-		stage5_dialog_post_midboss();
+TASK(loop_swirl, {
+	cmplx start;
+	cmplx velocity;
+	cmplx turn;
+	cmplx retention;
+}) {
+	Enemy *e = TASK_BIND(espawn_swirl(ARGS.start, ITEMS(.points = 4, .power = 2)));
 
-		// TODO: determine if this hack is still required with the new dialogue system
-		global.timer++;
+    e->move = move_linear(ARGS.velocity);
+    INVOKE_SUBTASK_DELAYED(80, loop_swirl_move,
+        .move = &e->move,
+		.turn = ARGS.turn,
+		.retention = ARGS.retention
+    );
+
+	int difficulty = 30 - difficulty_value(4, 8, 16, 24);
+
+	for (int x = 0; x < 200; x++) {
+		for(int i = 1; i >= -1; i -= 2) {
+			PROJECTILE(
+				.proto = pp_bullet,
+				.pos = e->pos,
+				.color = RGB(0.3, 0.4, 0.5),
+				.move = move_asymptotic_simple(i * 2 * ARGS.velocity * I / cabs(ARGS.velocity), 3),
+				// TODO: bullets only fire down-left with the above
+				// they should also fire in other directions correlating to the angle of "turn" the swirls make
+				//.rule = asymptotic,
+				//.args = { i*2*e->args[0]*I/cabs(e->args[0]), 3 }
+			);
+		}
+
+		play_sound("shot1");
+		WAIT(difficulty);
 	}
+}
 
-	FROM_TO(3000, 3200, 100) {
-		create_enemy1c(VIEWPORT_W/2 + VIEWPORT_W/6*(1-2*(_i&1)), 2000, BigFairy, stage5_lightburst2, -1+2*(_i&1) + 2.0*I);
+TASK(loop_swirls, {
+	int num;
+	int direction;
+	cmplx start;
+	cmplx velocity;
+	cmplx turn;
+	cmplx retention;
+}) {
+	for (int i = 0; i < ARGS.num; i++) {
+		INVOKE_TASK(loop_swirl, {
+			.start = VIEWPORT_W * ARGS.direction + ARGS.start * rng_sreal(),
+			.velocity = ARGS.velocity,
+			.turn = ARGS.turn * rng_sreal() + ARGS.start,
+			.retention = ARGS.retention,
+		});
+		WAIT(10);
 	}
+}
+
+TASK(limiter_fairy, {
+	cmplx pos;
+	MoveParams exit;
+}) {
+	Enemy *e = TASK_BIND(espawn_fairy_red(ARGS.pos, ITEMS(.points = 4, .power = 4)));
+	e->move = ARGS.exit;
+
+	int difficulty = difficulty_value(0.25, 0.50, 0.75, 1);
+	for (int x = 0; x < 400; x++) {
+		double base_angle = carg(global.plr.pos - e->pos);
 
-	FROM_TO(3300, 4000, 90-10*global.diff)
-		create_enemy1c(200.0*I+VIEWPORT_W*(_i&1), 1500, Fairy, stage5_superbullet, 3-6*(_i&1));
+		for (int i = 1; i >= -1; i -= 2) {
+			double a = i * 0.2 - 0.1 * difficulty + i * 3.0 / (x + 1);
+			cmplx aim = cdir(base_angle + a);
```

Comment:
1. Don't use `double`, use `real`
2. Better store the aiming direction vector e.g. `cmplx plraim = cnormalize(global.plr.pos - e->pos)`. Then you can rotate it by an arbitrary angle with `cmplx aim = plraim * cdir(a)`